Using static libraries may result in object files overwriting each other

Ideally, we'd like to name the files implementing the hardware driver for a particular board the same as the board itself. In most cases, this is easy. In case of the Keyboardio Atreus, there's a slight problem: we already have the original Atreus (`Technomancy/Atreus`), and if we rename the files from `Keyboardio/Atreus2.h` (and `.cpp`) to `Keyboardio/Atreus.h`, we get linker errors:

```
libraries/Kaleidoscope/Kaleidoscope.a(Runtime.cpp.o): In function `kaleidoscope::driver::keyscanner::ATmega<kaleidoscope::device::keyboardio::AtreusProps::KeyScannerProps>::actOnMatrixScan()':
/Users/jesse/Documents/GitHub/Kaleidoscope-Bundle-Keyboardio/avr/libraries/Kaleidoscope/src/kaleidoscope/Runtime.cpp:26: undefined reference to `kaleidoscope::driver::keyscanner::ATmega<kaleidoscope::device::keyboardio::AtreusProps::KeyScannerProps>::previousKeyState_'
```

...and so on.

This appears to happen because we're using `dot_a_linkage=true` in `library.properties`, the problem goes away if we set that to false. This suggests that we have a naming conflict, that in the static archive, `Technomancy/Atreus.o` (which is empty when we're compiling a sketch for the Keyboardio Atreus) ends up overwriting `Keyboardio/Atreus.o`.

There are a number of `Base` classes in the driver namespace, which one would expect to conflict as well, but they don't. They do not for the simple reason that they are header-only, and as such, do not have an object file compiled.

As a workaround - if we can't ask `ar` to store relative paths for us - would be to alter the build rules so that we produce `path/to/path_to_stuff.o` files instead of `path/to/stuff.o`, thus avoiding the conflict.

At the moment, this only affects the Keyboardio Atreus, but may become the problem down the road, so we should eventually figure something out.

--- a/makefiles/rules.mk
+++ b/makefiles/rules.mk
@@ -24,7 +24,7 @@ all: build-all
 astyle:
 	PATH="$(PLUGIN_TEST_BIN_DIR):$(PATH)" $(PLUGIN_TEST_SUPPORT_DIR)/quality/run-astyle
 
-travis-test: find-duplicates travis-smoke-examples travis-check-astyle shellcheck
+travis-test: find-filename-conflicts travis-smoke-examples travis-check-astyle shellcheck
 
 test: smoke-examples check-astyle cpplint check-docs
 
@@ -50,9 +50,9 @@ shellcheck:
 		shellcheck ${SHELL_FILES}; \
 	fi
 
-find-duplicates:
+find-filename-conflicts:
 	@if [ -d "bin" ]; then \
-		bin/find-duplicate-cpp-files; \
+		bin/find-filename-conflicts; \
 	fi
 
 ## NOTE: HERE BE DRAGONS, DO NOT CLEAN THIS UP!

--- a/makefiles/rules.mk
+++ b/makefiles/rules.mk
@@ -24,7 +24,7 @@ all: build-all
 astyle:
 	PATH="$(PLUGIN_TEST_BIN_DIR):$(PATH)" $(PLUGIN_TEST_SUPPORT_DIR)/quality/run-astyle
 
-travis-test: travis-smoke-examples travis-check-astyle shellcheck
+travis-test: find-duplicates travis-smoke-examples travis-check-astyle shellcheck
 
 test: smoke-examples check-astyle cpplint check-docs
 
@@ -48,6 +48,11 @@ cpplint:
 shellcheck:
 	@if [ -d "bin" ]; then \
 		shellcheck ${SHELL_FILES}; \
+	fi
+
+find-duplicates:
+	@if [ -d "bin" ]; then \
+		bin/find-duplicate-cpp-files; \
 	fi
 
 ## NOTE: HERE BE DRAGONS, DO NOT CLEAN THIS UP!

